src/IteratorToArrayConvertCapable.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Util\Normalization\NormalizeIterableCapableTrait;
+use stdClass;
+use Traversable;
+
+/**
+ * Trait IteratorToArrayConvertCapable, adds ability to convert an iterator to an array.
+ *
+ * @since [*next-version*]
+ */
+trait IteratorToArrayConvertCapable
+{
+    /* @since [*next-version*] */
+    use NormalizeIterableCapableTrait;
+
+    /**
+     * Convert an iterator to an array.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $iterator    The object to convert.
+     * @param callable                   $transformer Optional function that applies to value.
+     *
+     * @throws InvalidArgumentException If $iterator is not iterable.
+     *
+     * @return array Result of iterator to array transform.
+     */
+    protected function _iteratorToArrayRecursive($iterator, $transformer = null)
```

Comment:
Try to follow the naming convention that we already use, where the trait is named after the main method that it provides. Example: `IteratorToArrayRecursiveCapableTrait`.